### ISSUE TITLE
[6.x] TS6133 false positive in Astro frontmatter

### ISSUE THREAD

**Issue by Yu-yu0202** [withastro/compiler-rs#49]
### Astro Info

```block
❯ bunx --bun astro info
Astro                    v6.4.8
Node                     v24.3.0
System                   Linux (x64)
Package Manager          bun
Output                   server
Adapter                  @astrojs/node
Integrations             @astrojs/react
```

### Describe the Bug

A function called from a top-level return statement in Astro frontmatter is incorrectly reported as unused (TS6133).

The function is referenced and called, but `astro check` reports:

'notFound' is declared but its value is never read. ts(6133)

Adding an additional reference such as `void notFound();` removes the diagnostic.

### What's the expected result?

No TS6133 diagnostic should be reported because the function is referenced and called.

### Link to Minimal Reproducible Example

https://stackblitz.com/edit/astro-repro-ts6133

### Participation

- [ ] I am willing to submit a pull request for this issue.

**Comment by Yu-yu0202** [comment]
I'm a Japanese speaker and not fluent in English. This report was translated using an LLM, so please excuse any awkward wording.

**Comment by matthewp** [comment]
No problem and thank you.

**Comment by matthewp** [comment]
@astrobot-houston Please triage this again as it failed to finish the first time.

**Comment by matthewp** [comment]
This is not the correct solution. This is a compiler bug. I believe we are going to update the compiler to have TSX output, in which case hopefully we can fix it at that point.
